spec/spec.md

```diff
@@ -405,19 +402,60 @@ be null, and its value MUST match the [`did` input field](#did) that was used wh
 
 #### `didState.secret`
 
-This output field contains DID controller keys and other secrets.
+This output field contains an object with DID controller keys and other secrets.
+
+It MUST be present if the DID Registrar is operating in [Internal Secret Mode](#internal-secret-mode) and the
+[`secretReturning` option](#secretreturning-option) is set to `true`, and MUST NOT be present otherwise.
+
+If present, this output field MUST contain a JWK Set (JWKS) according to [[spec:RFC7517]]. The `kid` parameters of the
```

Comment:
Encrypted JWKS is a JWE (https://datatracker.ietf.org/doc/html/rfc7517#section-7)
Should the secret response be an object, that contains the following claims:
- type: a string; one of: [jwks, jwe]
- payload: JWKS object if type is jwks or serialzied JWK as string if type is jwe